pkg/aggregator/demultiplexer_agent.go

```diff
@@ -426,47 +427,27 @@ func (d *AgentDemultiplexer) flushToSerializer(start time.Time, waitForSerialize
 						blockChan:         make(chan struct{}),
 						waitForSerializer: waitForSerializer,
 					},
-					flushedSketches: &flushedSketches,
-					seriesSink:      seriesSink,
+					sketchesSink: sketchesSink,
+					seriesSink:   seriesSink,
 				}
 
 				d.aggregator.flushChan <- t
 				<-t.trigger.blockChan
 			}
 		}, func(serieSource metrics.SerieSource) {
 			sendIterableSeries(d.sharedSerializer, start, serieSource)
+		},
+		func(sketches metrics.SketchesSource) {
+			// Don't send empty sketches payloads
```

Comment:
I don't think we need to add a check for this case but I have no strong opinion on that.

Reply:
I created `sendIterableSeries` because it was used at several locations. The behavior of sketches is not the same in https://github.com/DataDog/datadog-agent/blob/12740f221d79ce56c39adf2650c6c9a10f1c0a71/pkg/aggregator/demultiplexer_serverless.go#L125-L127 versus https://github.com/DataDog/datadog-agent/blob/12740f221d79ce56c39adf2650c6c9a10f1c0a71/pkg/aggregator/demultiplexer_agent.go#L441-L448